include/api.php

```diff
@@ -2362,29 +2362,54 @@ function api_format_items_activities(&$item, $type = "json") {
 			'attendno' => array(),
 			'attendmaybe' => array()
 		);
+
 		$items = q('SELECT * FROM item
 					WHERE uid=%d AND `thr-parent`="%s" AND visible AND NOT deleted',
 					intval($item['uid']),
 					dbesc($item['uri']));
+
 		foreach ($items as $i){
-			builtin_activity_puller($i, $activities);
+			// not used as result should be structured like other user data
+			//builtin_activity_puller($i, $activities);
+
+			// get user data and add it to the array of the activity
+			$user = api_get_user($a, $i['author-link']);			
+			switch($i['verb']) {
+				case ACTIVITY_LIKE:
+					$activities['like'][] = $user;
+					break;
+				case ACTIVITY_DISLIKE:
+					$activities['dislike'][] = $user;
+					break;
+				case ACTIVITY_ATTEND:
+					$activities['attendyes'][] = $user;
+					break;
+				case ACTIVITY_ATTENDNO:
+					$activities['attendno'][] = $user;
+					break;
+				case ACTIVITY_ATTENDMAYBE:
+					$activities['attendmaybe'][] = $user;
+					break;
+				default:
+					break;
+			}
 		}
 
 		if ($type == "xml") {
 			$xml_activities = array();
-			foreach ($activities as $k => $v)
+			foreach ($activities as $k => $v) {
+				// change xml element from "like" to "friendica:like"
 				$xml_activities["friendica:".$k] = $v;
-
+				// add user data into xml output
+				$k_user = 0;
+				foreach ($v as $user)
+					$xml_activities["friendica:".$k][$k_user++.":user"] = $user;
```

Comment:
yeah, this is the magic line which made it working. How does it works? I really don't know! One would expect <k_user:user> but it returns <user>. The XML output now looks like this 
`<friendica:activities>
   <friendica:like>
      <user>
         <id>x</id>
         ...
         <network>dfrn</network>
      </user>
      <user>
         <id>x</id>
         ...
         <network>dfrn</network>
      </user>
   </friendica:like>
   <friendica:dislike>
      <user> ... </user>
   </friendica:dislike>
   <friendica:attendyes/>
   <friendica:attendno/>
   <friendica:attendmaybe/>
</friendica:activities>`
This seems to me as a correct xml syntax. \<user\> should be ok for me with my limited knowledge about xml, as it contains the same information as the \<user\> field has in Twitter API (no namespace => xmlns="http://api.twitter.com"). But if you prefer, we can change this to \<friendica:user\>. It doesn't matter for me as I am using json only for the Windows10 client.